src/ln/channelmonitor.rs

```diff
@@ -1471,107 +1475,115 @@ impl ChannelMonitor {
-								// payment_preimage.
-								// Note that if the dust limit were allowed to change between
-								// commitment transactions we'd want to be check whether *any*
-								// broadcastable commitment transaction has the HTLC in it, but it
-								// cannot currently change after channel initialization, so we don't
-								// need to here.
-								for &(ref broadcast_htlc, ref broadcast_source) in per_commitment_data.iter() {
-									if broadcast_htlc.transaction_output_index.is_some() && Some(source) == broadcast_source.as_ref() {
-										continue $id;
-									}
-								}
-								log_trace!(self, "Failing HTLC with payment_hash {} from {} remote commitment tx due to broadcast of remote commitment transaction", log_bytes!(htlc.payment_hash.0), $commitment_tx);
-								match self.onchain_events_waiting_threshold_conf.entry(height + ANTI_REORG_DELAY - 1) {
-									hash_map::Entry::Occupied(mut entry) => {
-										let e = entry.get_mut();
-										e.retain(|ref event| {
-											match **event {
-												OnchainEvent::HTLCUpdate { ref htlc_update } => {
-													return htlc_update.0 != **source
-												},
-												_ => return true
-											}
-										});
-										e.push(OnchainEvent::HTLCUpdate { htlc_update: ((**source).clone(), htlc.payment_hash.clone())});
-									}
-									hash_map::Entry::Vacant(entry) => {
-										entry.insert(vec![OnchainEvent::HTLCUpdate { htlc_update: ((**source).clone(), htlc.payment_hash.clone())}]);
-									}
-								}
-							}
-						}
-					}
-				}
-			}
-			if let Storage::Local { ref current_remote_commitment_txid, ref prev_remote_commitment_txid, .. } = self.key_storage {
-				if let &Some(ref txid) = current_remote_commitment_txid {
-					check_htlc_fails!(txid, "current", 'current_loop);
-				}
-				if let &Some(ref txid) = prev_remote_commitment_txid {
-					check_htlc_fails!(txid, "previous", 'prev_loop);
-				}
-			}
+		} else if let Some(revocation_points) = self.their_cur_revocation_points {
+			let revocation_point_option =
+				if revocation_points.0 == commitment_number { Some(&revocation_points.1) }
+				else if let Some(point) = revocation_points.2.as_ref() {
+					if revocation_points.0 == commitment_number + 1 { Some(point) } else { None }
+				} else { None };
+			if let Some(revocation_point) = revocation_point_option {
+
+				// While this isn't useful yet, there is a potential race where if a counterparty
+				// revokes a state at the same time as the commitment transaction for that state is
+				// confirmed, and the watchtower receives the block before the user, the user could
+				// upload a new ChannelMonitor with the revocation secret but the watchtower has
+				// already processed the block, resulting in the remote_commitment_txn_on_chain entry
+				// not being generated by the above conditional. Thus, to be safe, we go ahead and
+				// insert it here.
+				watch_outputs.append(&mut tx.output.clone());
```

Comment:
Huh? Now we just watch everything? This still needs to be filtered.

Reply:
Yes right we don't care to watch how to_remote output is spent and others outputs only if they are htlcs on remote valid commitment transaction. Moved it beyond Some(per_commitment_data) = per_commitment_option branch